### DISCUSSION TITLE
Picture partial is very slow

### DISCUSSION THREAD

**Discussion by tobimori** [studio1902/statamic-peak#354] [Q&A]
I have a lot of images on my page, often multiple ones per page builder block. 

Most of my loading times comes from using the Peak picture partial, with single ones taking up to half a second of loading time.
![CleanShot 2023-10-27 at 17 52 28@2x](https://github.com/studio1902/statamic-peak/assets/29142128/76a6cc86-cd2b-4773-a8c5-3bb45d3ff28a)

![CleanShot 2023-10-27 at 17 56 17@2x](https://github.com/studio1902/statamic-peak/assets/29142128/f928d09c-1ec9-43c6-aaa6-ca0d0f62a51c)

![CleanShot 2023-10-27 at 17 58 12@2x](https://github.com/studio1902/statamic-peak/assets/29142128/59b32600-fd43-4f23-8c68-5ca7d80bd513)

Any ideas on how to improve the performance?

**Comment by robdekort** [comment]
The partial definitely is a heavy one, but performance doesn't seem nearly as slow on my local dev machine or production. I typically use static caching which makes render time a non issue. You could also use cache tags when using the partial. 

One thought: could it be you have `save_cached_images` turned to `true` and it takes the generation of all images into account with rendering time?

**Reply by tobimori** [reply]
Yes, I use Glide caching as otherwise images took ages to load.
I also use Static Caching, but this doesn't work for all pages, for example the custom 404 page. Mine takes around 2,5 secs to load with navigation, footer, and one page builder block that has around 20 images. Even with cached images, it doesn't get faster.

**Reply by robdekort** [reply]
Despite the name, `save_cached_images` doesn't do what you expect: https://wentsch.me/what-statamics-save_cached_images-really-does/. I've find turning it off really improves initial page loads as the first request doesn't generate all images in the sourceset, but only the ones currently needed. Plus it doesn't stop rendering the page. Besides this I have no other ideas.

**Reply by tobimori** [reply]
I'll check what changing that does, thanks.

**Reply by robdekort** [reply]
Cool, hope it helps a bit. Keep us posted.

**Reply by tobimori** [reply]
Turns out I didn't have it activated on development which the screenshots above are from.

**Reply by robdekort** [reply]
Too bad.

**Reply by tobimori** [reply]
![CleanShot 2023-10-28 at 13 37 45@2x](https://github.com/studio1902/statamic-peak/assets/29142128/407b99b1-f056-4189-bc59-a89439364f0b)

This is the result of disabling it on production. All images fail to load.

**Reply by robdekort** [reply]
Must be the result of something else. I don't get this.

**Reply by robdekort** [reply]
Perhaps you need to clear the glide cache first.

**Reply by tobimori** [reply]
Glide crashed the PHP server

**Reply by robdekort** [reply]
I feel this all extended Peak support by now though. Good luck.

**Reply by tobimori** [reply]
No worries. It seems like it was just firing to many requests at once, meaning PHP wasn't allowed to start more workers. I increased the `max_children` setting and it seems to work fine for now.

Although uncached glide requests now take 2-4s usually, do you know if there's a way to speed that up?

**Reply by robdekort** [reply]
I don't think you can. It has to process those images.

**Reply by tobimori** [reply]
I see, have to investigate further.

**Reply by robdekort** [reply]
Yeah, did that at some point as well. And then figured out some great defaults that work perfectly in combination with static caching for me. Those are the Peak defaults as well btw.

**Reply by tobimori** [reply]
Do you know how we could improve the performance of the 404 entry rendering which isn't cached?

**Reply by robdekort** [reply]
Do you only use the picture partial or all of Peak? Because I _think_ in Peak it does get cached, because I use an actual entry for the 404 page.

**Reply by tobimori** [reply]
I do use the 404 thing from Peak as well

**Reply by tobimori** [reply]
Okay, I just wrapped the 404 templtate in a cache tag and that seems to speed it up.

**Reply by tobimori** [reply]
thanks for your help!

**Comment by eliasu** [comment]
> Most of my loading times comes from using the Peak picture partial, with single ones taking up to half a second of loading time.
> ![CleanShot 2023-10-27 at 17 52 28@2x](https://private-user-images.githubusercontent.com/29142128/278695867-76a6cc86-cd2b-4773-a8c5-3bb45d3ff28a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkxMjYyMTYsIm5iZiI6MTcwOTEyNTkxNiwicGF0aCI6Ii8yOTE0MjEyOC8yNzg2OTU4NjctNzZhNmNjODYtY2QyYi00NzczLWE4YzUtM2JiNDVkM2ZmMjhhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI4VDEzMTE1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY4NTAxODJlNGJlNzNmZWIxZmM0NGQyMjE4M2U1NmUxOTY2MjA5OTljZWM0ZjBkZjUyODc0ZTFkMTI0NzAwMTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5C-JruOXn-m0nKQ_y9sW5QNsUWr5jJt_XOLkboYRrk8)

Hey, I just saw your post, and I was wondering where you can debug the times of tags? Is it in Chrome or where do you get this view?

**Reply by tobimori** [reply]
This is the Debugbar from Statamic

https://statamic.dev/debugging#debug-bar
